lib/modules/manager/deno/schema.ts

```diff
@@ -0,0 +1,429 @@
+          versioning: denoVersioningId,
+        };
+
+        if (depName && !validateNpmPackageName(depName).validForOldPackages) {
+          dep.skipReason = 'invalid-name';
+          return dep;
+        }
+
+        if (currentValue && !isValid(currentValue)) {
+          dep.skipReason = 'invalid-version';
+          return dep;
+        }
+        return {
+          ...dep,
+          depName,
+          currentValue,
+          depType,
+          currentRawValue,
+        };
+      }
+      // jsr datasource
+      if (datasource === 'jsr') {
+        const dep: PackageDependency = {
+          datasource,
+          versioning: denoVersioningId,
+        };
+
+        if (depName && !extractJsrPackageName(depName)) {
+          dep.skipReason = 'invalid-name';
+          return dep;
+        }
+
+        if (currentValue && !isValid(currentValue)) {
+          dep.skipReason = 'invalid-version';
+          return dep;
+        }
+        return {
+          ...dep,
+          depName,
+          currentValue,
+          depType,
+          currentRawValue,
+        };
+      }
+    }
+
+    return {
+      depType,
+      depName: depValue,
+      skipStage: 'extract',
+      skipReason: 'unsupported',
+    };
+  });
+
+export const ImportMap = Json.pipe(
+  z.object({
+    imports: z.optional(Imports).default({}),
+    scopes: z.optional(Scopes).default({}),
+  }),
+);
+export type ImportMap = z.infer<typeof ImportMap>;
+
+// https://github.com/denoland/deno/blob/410c66ad0d1ce8a5a3b1a2f06c932fb66f25a3c6/cli/schemas/config-file.v1.json
+export const DenoPackageFile = z
+  .object({
+    lock: z
+      .union([
+        z.string(),
+        z.boolean(),
+        z.object({
+          path: z.string().optional(),
+        }),
+      ])
+      .optional(),
+    imports: z.optional(Imports).default({}),
+    scopes: z.optional(Scopes).default({}),
+    importMap: z.string().optional(),
+    tasks: z.optional(Tasks).default({}),
+    compilerOptions: z.optional(CompilerOptions).default({}),
+    lint: z.optional(Lint).default({}),
+    workspace: Workspace,
+  })
+  .transform(
+    ({
+      lock,
+      imports,
+      scopes,
+      tasks,
+      compilerOptions,
+      lint,
+      importMap,
+      workspace,
+    }) => ({
+      lock,
+      importMap,
+      managerData: {
+        workspaces: workspace?.workspaces,
+      },
+      dependencies: [
+        ...imports,
+        ...scopes,
+        ...tasks,
+        ...compilerOptions,
+        ...lint,
+      ],
+    }),
+  );
+export type DenoPackageFile = z.infer<typeof DenoPackageFile>;
+
+export const DenoExtract = z
+  .object({
+    content: Jsonc.pipe(DenoPackageFile),
+    fileName: z.string(),
+  })
+  .transform(async (data) => {
+    // handle lock file
+    const { content, fileName } = data;
+
+    let lockFile: string | undefined;
+    if (content.lock) {
+      const lock = content.lock;
+      if (typeof lock === 'string' && (await localPathIsFile(lock))) {
```

Comment:
This schema must be just a function (because `async` transforms are potentially problematic). However, this `typeof` cascade indicates the lack of schema somewhere else.